app/experimenter/nimbus-ui/src/components/PageHome/FilterBar/index.tsx

```diff
@@ -0,0 +1,133 @@
+
+import React, { useMemo } from "react";
+import Nav from "react-bootstrap/Nav";
+import Navbar from "react-bootstrap/Navbar";
+import Select, { OptionsType, OptionTypeBase } from "react-select";
+import { NullableObjectArray } from "../../../lib/types";
+import { FilterOptions, FilterValue } from "../types";
+
+export type FilterBarProps = {
+  options: FilterOptions;
+  value: FilterValue;
+  onChange: (value: FilterValue) => void;
+};
+
+export const FilterBar: React.FC<FilterBarProps> = ({
+  options,
+  value: filterValue,
+  onChange,
+}) => {
+  return (
+    <Navbar variant="light" bg="light" className="nav-fill mt-4 mb-4">
+      <Nav className="w-100">
+        <FilterSelect
+          fieldLabel="Feature"
+          fieldOptions={options.featureConfig!}
+          filterValueName="featureConfig"
+          optionLabelName="name"
+          optionValueName="slug"
+          {...{ filterValue, onChange }}
+        />
+        <FilterSelect
+          fieldLabel="Application"
+          fieldOptions={options.application!}
+          filterValueName="application"
+          optionLabelName="label"
+          optionValueName="value"
+          {...{ filterValue, onChange }}
+        />
+        <FilterSelect
+          fieldLabel="Owner"
+          fieldOptions={options.owners!}
+          filterValueName="owners"
+          optionLabelName="username"
+          optionValueName="username"
+          {...{ filterValue, onChange }}
+        />
+        <FilterSelect
+          fieldLabel="Version"
+          fieldOptions={options.firefoxMinVersion!}
+          filterValueName="firefoxMinVersion"
+          optionLabelName="label"
+          optionValueName="value"
+          {...{ filterValue, onChange }}
+        />
+      </Nav>
+    </Navbar>
+  );
+};
+
+type FilterSelectProps<T extends NullableObjectArray> = {
+  filterValue: FilterValue;
+  onChange: (value: FilterValue) => void;
+  filterValueName: keyof FilterOptions;
+  fieldLabel: string;
+  fieldOptions: T;
+  optionLabelName: keyof NonNullable<T[number]>;
+  optionValueName: keyof NonNullable<T[number]>;
+};
+
+const FilterSelect = <T extends NullableObjectArray>({
+  filterValue,
+  onChange,
+  filterValueName,
+  fieldLabel,
+  fieldOptions,
+  optionLabelName,
+  optionValueName,
+}: FilterSelectProps<T>) => {
+  const filterOptions = useMemo(
+    () =>
+      fieldOptions.filter(
+        (option): option is NonNullable<T[number]> => option !== null,
+      ),
+    [fieldOptions],
+  );
+  const fieldValue = filterValue[filterValueName];
+  const selectedOptions = useMemo(
+    () =>
+      filterOptions.filter(
+        (option) =>
+          typeof option[optionValueName] === "string" &&
+          fieldValue?.includes("" + option[optionValueName]),
+      ),
+    [fieldValue, filterOptions],
+  );
+
+  return (
+    <Nav.Item
+      className="m-1 text-left flex-grow-1 flex-shrink-1"
+      style={{ flexBasis: "0px" }}
```

Comment:
I think instead of flex-grow/flex-shrink + flex-basis, we only need `flex-basis: 1`. Can we add a custom class in `styles/index.scss` and maybe call it `flex-1`? (hmm, that SCSS file is starting to feel a bit messy)

Reply:
Why are we using flex explicitly instead of using `Row` and `Col` to do columnar layout?

Reply:
Used these specific class names after wrestling with row, col, and other flex properties. This was the only combination that seemed to land on the exact semi-fixed width behavior we want for the selectors. Could be worth continuing to poke at, but maybe in a follow up